crates/matrix-sdk-ui/src/unable_to_decrypt_hook.rs

```diff
@@ -63,6 +63,21 @@ pub struct UnableToDecryptInfo {
     /// What we know about what caused this UTD. E.g. was this event sent when
     /// we were not a member of this room?
     pub cause: UtdCause,
+
+    /// The difference between the event creation time (`origin_server_ts`) and
+    /// the time our device was created. If negative, this event was sent
+    /// *before* our device was created.
+    pub event_local_age_millis: i64,
+
+    /// Whether the user had verified their own identity at the point they
+    /// received the UTD event.
+    pub user_trusts_own_identity: bool,
+
+    /// True if the sender of the event was on a different server to our user.
+    pub is_federated: bool,
+
+    /// True if our user is using the `matrix.org` homeserver.
+    pub is_matrix_dot_org: bool,
```

Comment:
It feels strange to have a `is_matrix_dot_org` at the `sdk` level. 
Can we have instead a sender_homerserver and a own_homeserver , then the app could do a  `equals("matrix.org")`  and the is_federated?

Reply:
I was torn on this. On the one hand, I agree: this does seem a bit arbitrary. On the other, I was keen to factor out a as much as possible of the EXA/EXI common code.

Reply:
I think @BillCarsonFr's idea is quite nice. I understand your position, but I think it's important to not take this kind of shortcuts/biases in the SDK.

Reply:
Fair enough. PTAL.